adam-core/src/main/scala/org/bdgenomics/adam/models/ReferenceRegion.scala

```diff
@@ -510,12 +560,30 @@ case class ReferenceRegion(
   }
 
   override def hashCode: Int = {
-    var result = 37
-    result = 41 * result + (if (referenceName != null) referenceName.hashCode else 0)
-    result = 41 * result + start.hashCode
-    result = 41 * result + end.hashCode
-    result = 41 * result + (if (strand != null) strand.ordinal() else 0)
-    result
+    // add 37 here because this is the start of the hashCode
+    val nameHashCode = 37 + {
```

Comment:
Reading the commit message, that method was necessary to fix issue #817, which was a regression caused by 9a318ba40527ad660dd7637d8a7d5536adf14c7e from PR #624. I would say it definitely was not a performance issue... If we do delete the hash code method, we need to validate that it doesn't cause #817 to reappear.